core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -710,7 +710,11 @@ private[spark] class TaskSetManager(
       logInfo(s"Killing attempt ${attemptInfo.attemptNumber} for task ${attemptInfo.id} " +
         s"in stage ${taskSet.id} (TID ${attemptInfo.taskId}) on ${attemptInfo.host} " +
         s"as the attempt ${info.attemptNumber} succeeded on ${info.host}")
-      sched.backend.killTask(attemptInfo.taskId, attemptInfo.executorId, true)
+      sched.backend.killTask(
+        attemptInfo.taskId,
+        attemptInfo.executorId,
+        interruptThread = true,
+        reason = "another attempt succeeded")
```

Comment:
I added two screenshots to the PR description. In the second scenario having a verbose reason is fine, but in the stage summary view long or many distinct reasons would overflow the progress bar.

We could probably fix the css to allow slightly longer / more reasons, but even that wouldn't be great if each task had a different reason.